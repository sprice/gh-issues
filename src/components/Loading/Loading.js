import React from "react"
import PropTypes from "prop-types"
import ViewerHeader from "../../components/ViewerHeader"

class Loading extends React.Component {
  render() {
    return (
      <div className="loading">
        <ViewerHeader githubUrl={this.props.githubUrl} />
        <div className="spinner">
          <div className="spin" />
        </div>
      </div>
    )
  }
}

Loading.propTypes = {
  githubUrl: PropTypes.string.isRequired
}

export default Loading
