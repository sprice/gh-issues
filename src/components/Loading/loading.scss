.spinner {
  width: 100vw;
  height: calc(100vh - 5rem);
  display: flex;
  justify-content: center;
  align-items: center;
}

.spin {
  background: url("../icons/spinner.svg") no-repeat;
  width: 100px;
  height: 100px;
}
