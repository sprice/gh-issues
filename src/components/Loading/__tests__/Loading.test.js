import React from "react"
import { create } from "react-test-renderer"
import Loading from "../Loading"

test("Snapshop of Loading", () => {
  const c = create(<Loading githubUrl="https://github.com/ethereum/soldity" />)
  expect(c.toJSON()).toMatchSnapshot()
})
