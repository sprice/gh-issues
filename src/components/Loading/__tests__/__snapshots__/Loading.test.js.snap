// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshop of Loading 1`] = `
<div
  className="loading"
>
  <header>
    <h2>
      Github Issue Viewer
    </h2>
    <h4
      className="regular"
    >
      https://github.com/ethereum/soldity
    </h4>
  </header>
  <div
    className="spinner"
  >
    <div
      className="spin"
    />
  </div>
</div>
`;
